From bugs+caf_=bugs=bugs.plt-scheme.org@plt-scheme.org Thu May  9 13:25:50 2013
Received: from mail-ob0-f175.google.com (mail-ob0-f175.google.com [209.85.214.175])
	by champlain.ccs.neu.edu (8.14.4/8.14.4) with ESMTP id r49HPlT9010137
	for <bugs@bugs.plt-scheme.org>; Thu, 9 May 2013 13:25:47 -0400
Message-Id: <201305091725.r49HPjX2010131@champlain.ccs.neu.edu>
Date: Thu, 9 May 2013 13:25:45 -0400
From: danking@ccs.neu.edu
To: bugs@racket-lang.org
Subject: XREPL becomes unusable if an error occurs while output is redirected to a file

>Number:         13744
>Category:       all
>Synopsis:       XREPL becomes unusable if an error occurs while output is redirected to a file
>Class:          sw-bug
>Responsible:    eli
>Severity:       serious
>Priority:       medium
>State:          closed
>Confidential:   no
>Arrival-Date:   Thu May 09 13:28:01 -0400 2013
>Closed-Date:    Mon Aug 19 02:37:28 -0400 2013
>Last-Modified:  Mon Aug 19 02:37:28 -0400 2013
>Originator:     Daniel King
>Organization:
plt
>Submitter-Id:   unknown
>Release:        5.3.2
>Environment:
Linux i686 / Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.17 (KHTML, like Gecko) Chrome/24.0.1312.57 Safari/537.17
>Description:
Run in an xrepl:

  (with-output-to-file "/tmp/foo" (lambda () (/ 1 0)))

Get an infinite loop printing

  exception raised by error display handler: flush-output: output port is closed; original exception raised: flush-output: output port is closed
>How-To-Repeat:
(require xrepl)
(with-output-to-file "/tmp/foo" (lambda () (/ 1 0)))
>Fix:
>Audit-Trail:
Responsible changed from "nobody" to "eli" by eli@racket-lang.org at Mon, 19 Aug 2013 02:37:28 -0400
Reason>>> A commit by eli@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/a2122418ae
State changed from "open" to "closed" by eli@racket-lang.org at Mon, 19 Aug 2013 02:37:28 -0400
Reason>>> A commit by eli@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/a2122418ae
  | Fix a use of `fresh-line'.
  | 
  | Needs to do it when the relevant stderr is in effect.
  | 
  | Fixes PR 13744, except that something like this is still bad:
  | 
  |   (with-output-to-file "/tmp/foo" #:exists 'append
  |     (lambda ()
  |       (parameterize ([current-error-port (current-output-port)])
  |         (/ 1 0))))
  | 
  | But I think that it's a bad idea for the error display handler to be
  | called with the locally set port -- described in PR 13974.

